core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -463,11 +463,7 @@ public ByteBuffer keyMetadata() {
 
   @Override
   public List<Long> splitOffsets() {
-    if (splitOffsetList == null && splitOffsets != null) {
-      this.splitOffsetList = ArrayUtil.toUnmodifiableLongList(splitOffsets);
-    }
-
-    return splitOffsetList;
+    return splitOffsets == null ? ImmutableList.of() : Longs.asList(splitOffsets);
```

Comment:
The intent of the original PR was to avoid the inefficient allocation of long primitives into a collection. The Guava wrapper should be much more efficient than the original allocation code.

Reply:
I think we might want to return null in the event there are no split offsets. That looks like the original behavior, and I think most of the checks are looking for Noel to decide between offset planning and basic split planning.

Reply:
Right, I missed, that, I made this change.